src/main/java/appeng/util/item/ItemList.java

```diff
@@ -63,6 +66,8 @@ public boolean isEmpty() {
 
     @Override
     public void add(final IAEItemStack itemStack) {
+        version.incrementAndGet();
```

Comment:
Is this here on purpose to crash on trying to add something, even it the collections is unmodified? Otherwise it might be an idea to move it behind the null check. Or even move it into the `create` part of `getOrCreateRecord`.